lib/elixir/lib/module/parallel_checker.ex

```diff
@@ -309,24 +309,38 @@ defmodule Module.ParallelChecker do
     Enum.flat_map(warnings, fn {module, warning, locations} ->
       message = module.format_warning(warning)
       diagnostics = Enum.map(locations, &to_diagnostic(message, &1))
-      log? and :elixir_errors.print_warning([message, ?\n, format_stacktraces(diagnostics)])
+      log? and print_warning(message, diagnostics)
       diagnostics
     end)
   end
 
-  defp format_stacktraces([diagnostic]) do
-    format_diagnostic_stacktrace(diagnostic)
+  defp print_warning(message, [diagnostic]) do
+    :elixir_errors.print_warning(message, diagnostic)
   end
 
-  defp format_stacktraces(diagnostics) do
+  defp print_warning(message, [diagnostic | rest]) do
+    total_locations = length(rest)
+
+    locations =
+      rest
+      |> Enum.with_index(1)
+      |> Enum.map(fn {%{stacktrace: [s]}, i} ->
+        if i == total_locations do
+          ["  ", Exception.format_stacktrace_entry(s)]
+        else
+          ["  ", Exception.format_stacktrace_entry(s), ?\n]
+        end
+      end)
+
+    locations_plural = (total_locations == 1 && "location") || "locations"
```

Comment:
```suggestion
    locations_plural = if total_locations == 1, do: "location", else: "locations"
```

We don't use the other style anywhere in Elixir. :)